### PR TITLE
 Add additional links to main README

### PR DESCRIPTION
**Pull request by alamb**
# Which issue does this PR close?

Closes https://github.com/apache/arrow-datafusion/pull/7061

# Rationale for this change
Inspired by @MrPowers's proposal in https://github.com/apache/arrow-datafusion/pull/7061, I think we can make it easier for people who land at the DataFusion README to find what they are looking for quickly. 

# What changes are included in this PR?
See rendered change here: https://github.com/alamb/arrow-datafusion/tree/alamb/readme#readme

Screenshot 2023-07-26 at 9 11 37 AM

Basically:
1. Add links in bullet list for efficient location
2. Remove outdated codecov badge that is visually distracting
3. Add direct link to known users

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Review by jackwener — APPROVED** [review]

**Comment by alamb** [comment]
Thanks for the review @jackwener  -- I plan to leave this open for another day or so in case anyone has additional comments
